### ISSUE TITLE
Transfer learning for flair with custom data

### ISSUE THREAD

**Issue by desusai7** [flairNLP/flair#1356]
I am trying to apply transfer learning to pre-trained ner model of sequence tagger model with some data which is of type shown below but when i tried to train using that data it says that it was bad epochs and the model performance became worst after training.
This was the data i used for training:
![Screenshot (24)](https://user-images.githubusercontent.com/48179357/72597777-7ad3b980-3934-11ea-9883-0c1b598e3c92.png)
This was the output before training:
![Screenshot (25)](https://user-images.githubusercontent.com/48179357/72597879-b1a9cf80-3934-11ea-80df-b75524834b6d.png)
This is how training is done:
![Screenshot (26)](https://user-images.githubusercontent.com/48179357/72597983-eb7ad600-3934-11ea-96a3-c6dd5dfe0821.png)
This was the output of training:
![Screenshot (27)](https://user-images.githubusercontent.com/48179357/72598050-09483b00-3935-11ea-96f2-fa8e1c26aa55.png)
This was output of the model after training:
![Screenshot (29)](https://user-images.githubusercontent.com/48179357/72598180-4f9d9a00-3935-11ea-8563-daef79771933.png)

So Can anyone please help me to check
i)if the training data is in correct format
ii)if there are any issues in code
iii)why there are bad epochs coming

**Comment by deepankar27** [comment]
@desusai7 would be interested to know the solution..

**Comment by desusai7** [comment]
@deepankar27  working on the same and if anything works for me i would post it here and i had tried training the model with around 200 utterances but it failed and can you please share your training data format you were using

**Comment by djstrong** [comment]
For some reason the model is not training well. Present all training code (and not as image), especially how `corpus` is loaded.
Maybe label `per` must be uppercased.

**Comment by desusai7** [comment]
@djstrong I would look into changing per to PER and the training code is as follows
from flair.data import Corpus
from flair.embeddings import TokenEmbeddings, WordEmbeddings, StackedEmbeddings,CharacterEmbeddings,FlairEmbeddings
from typing import List
from flair.datasets import ColumnCorpus
from flair.models import SequenceTagger
from flair.trainers import ModelTrainer
columns = {0: 'text', 1: 'ner'}
data_folder = '/code'
corpus: Corpus = ColumnCorpus(data_folder, columns,train_file='train.csv',dev_file='dev.csv',test_file='test.csv')
tag_type = 'ner'
tag_dictionary = corpus.make_tag_dictionary(tag_type=tag_type)
embedding_types: List[TokenEmbeddings] = [  WordEmbeddings('glove'), CharacterEmbeddings(),FlairEmbeddings('news-forward'),FlairEmbeddings('news-backward'),]
embeddings: StackedEmbeddings = StackedEmbeddings(embeddings=embedding_types)

tagger = SequenceTagger(hidden_size=256,embeddings=embeddings,tag_dictionary=tag_dictionary,tag_type=tag_type,use_crf=True).load('ner')
trainer=ModelTrainer(tagger, corpus)
trainer.train('resources/taggers/example-ner',learning_rate=0.1,mini_batch_size=32,max_epochs=150)
model = SequenceTagger.load('resources/taggers/example-ner/best-model.pt')
sentence = Sentence('saivenkat')
model.predict(sentence)
print(sentence.to_tagged_string())

**Comment by severinsimmler** [comment]
@desusai7 is right. The original model is trained on _uppercase_ labels, your training data is partially _lowercase_. Make the labels in your custom training data uppercase and it should work.

**Comment by desusai7** [comment]
Thank you @djstrong  for your valuable suggestion and @severinsimmler  it worked brilliantly on the sentences which are ending with the name like "I am sreya" but the thing is when i tried to predict names in a sentence like "I am sreya and i am from hyderabad" then it is not predicting the label.
Do i have to change the training data into sentences format or the one which i am following is enough?

**Comment by severinsimmler** [comment]
Sorry, just looked at your training data, which basically does not make sense at all. Is this just a list of names? You do not need deep learning for this, you could just make a lookup-table or something. However, training data should be syntactically and semantically correct sentences with annotated named entities.

**Comment by desusai7** [comment]
@severinsimmler  I wanted to retrieve name ,age,gender and other personal information of a person from a raw text which does not have any syntactic and semantic meaning. So can you please suggest me what to follow ? Thank you :)

**Comment by severinsimmler** [comment]
If you do not have valid running text, I'd say sequence tagging with deep learning is not the way to go, if deep learning at all. [Wikidata](https://www.wikidata.org) as a knowledge base and a simple rule-based system might be an interesting starting point. If you have `hyderabad` and want to know what kind of entity it is, you could query Wikidata and get [Hyderabad](https://www.wikidata.org/wiki/Q1361)'s page. Now you have a lot structured information about Hyderabad and can extract in a deterministic way what you want to know about it.

**Comment by desusai7** [comment]
Thanks a lot. I would work on it.

**Comment by severinsimmler** [comment]
> Probably, the model have overfitted and forgot everything.

I don't think so; the model didn't overfit, it didn't learn anything at all. The task is sequence tagging, i. e. the tag of a token depends on the sequence it is part of. His training data has no sequence information at all, it's just a list of names or places or whatever.
Furthermore, the model he tries to fine-tune depends on Flair embeddings which are context-sensitive, so a vector representation of a token depends on the other tokens in the sequence. Another fact that a model like this cannot learn _anything_ from the training data.

**Comment by djstrong** [comment]
I disagree. It can be validated by training log and `monitor_train = True`, but I will not do it.
